### DISCUSSION TITLE
Update input value of field on focus

### DISCUSSION THREAD

**Discussion by MaartenJC** [filamentphp/filament#5781] [Help] [answered]
I can't get the following functionality working in Filament.
When the user foceses on an input or textarea, add a string to the value with the current timestamp and initials.
So i need a javascript function to be executed onFocus.
The old code works like this:
@focus="helpers.addTimestamp($event, 'MC')"

I tried it in Filament with the following code:
`->extraAttributes(['onFocus' => "helpers.addTimestamp(event, 'MC')"]);`

But the quotes get escaped like this:
`onfocus="helpers.addTimestamp(event, &#039;MC&#039;)"`

Or perhaps i'm looking in the wrong direction? Could it be done with a Livewire function?

**Comment by Sicklou** [comment]
With Alpine directive "on" you could maybe do it.

https://alpinejs.dev/directives/on

**Reply by MaartenJC** [reply]
Thanks for the suggestion.
But this doesn't solve the problem that the data in the function extraAttributes gets escaped.

**Comment by Sicklou** [comment, marked as answer]
Try like this : 
```php
->extraAttributes([
    'x-on:focus' => new HtmlString("console.log('test')")
])
```

**Reply by MaartenJC** [reply]
That worked, thank you so much!
